### PR TITLE
David freitag

### PR DESCRIPTION
**Pull request by freitagdavid**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,14 +1,96 @@
...
+            res.status(200).json({ zoos });
+        })
+        .catch(err => {
+            res.status(500).json(err);
+        });
+});
+
+server.post('/api/zoos', (req, res) => {
+    db('zoos')
+        .insert(req.body)
```
Comment (johnoro) [inline index.js:27]: You could destructure `body` here (and elsewhere) to be explicit in what you need off of the request object:
```javascript
server.post('/api/zoos', ({ body }, res) => {
    db('zoos')
        .insert(body)
```

File: index.js [hunk 2]
```diff
@@ -1,14 +1,96 @@
...
+        .then(result => {
+            if (result.length === 0) {
+                zooId404(res);
+            } else {
+                res.status(200).json(result);
+            }
+        })
+        .catch(err => {
+            res.status(500).json(err);
+        });
```
Comment (johnoro) [inline index.js:49]: It's probably be worth it to have a server error helper function:
```javascript
const servErr = res => err => {
  res.status(500).json(err);
};
```
and usage (in all of your `.catch`s):
```javascript
.catch(servErr(res));
```

File: index.js [hunk 3]
```diff
@@ -1,14 +1,96 @@
...
-
+const knex = require('knex');
+const dbConfig = require('./knexfile');
+const { zooId404 } = require('./helpers/error');
 const server = express();
 
 server.use(express.json());
 server.use(helmet());
 
+const db = knex({ ...dbConfig.development });
```
Comment (johnoro) [inline index.js:11]: Why use spread here? I haven't seen this before so I don't know if there's a reason to get a clean copy of the development object. This is how I've seen it in the past, though:
```javascript
const db = knex(dbConfig.development);

```

**Review by johnoro — APPROVED** [review]
Looking good!

File: index.js (hunk 3 above)
Comment (freitagdavid) [reply index.js:11]: Hrm yeah probably unnecessary. Not sure what I was thinking.

**Pull request merged**